src/solver/conv_ck_igemm_grp_fwd_bias_activ_fused.cpp

```diff
@@ -0,0 +1,504 @@
+
+        // in_strides  = {C, Hi * Wi * G * C, 1, Wi * G * C, G * C};
+        // out_strides = {K, Ho * Wo * G * K, 1, Wo * G * K, G * K};
+        // wei_strides = {K * Y * X * C, Y * X * C, 1, X * C, C};
+        auto miopen_in_strides  = problem.GetIn().GetStrides();
+        auto miopen_out_strides = problem.GetOut().GetStrides();
+        auto miopen_wei_strides = problem.GetWeights().GetStrides();
+        miopen_in_strides.insert(miopen_in_strides.begin(), C);
+        miopen_out_strides.insert(miopen_out_strides.begin(), K);
+        miopen_wei_strides.insert(miopen_wei_strides.begin(), K * miopen_wei_strides[0]);
+        std::copy(miopen_in_strides.begin(), miopen_in_strides.end(), in_strides.begin());
+        std::copy(miopen_out_strides.begin(), miopen_out_strides.end(), out_strides.begin());
+        std::copy(miopen_wei_strides.begin(), miopen_wei_strides.end(), wei_strides.begin());
+
+        strides  = {ProblemInterpreter::GetAdjustedConvolutionStrideH(problem),
+                   ProblemInterpreter::GetAdjustedConvolutionStrideW(problem)};
+        dilation = {ProblemInterpreter::GetAdjustedConvolutionDilationH(problem),
+                    ProblemInterpreter::GetAdjustedConvolutionDilationW(problem)};
+        lPadding = {ProblemInterpreter::GetInputLeftPadH(problem),
+                    ProblemInterpreter::GetInputLeftPadW(problem)};
+        rPadding = {ProblemInterpreter::GetAdjustedInputRightPadH(problem),
+                    ProblemInterpreter::GetAdjustedInputRightPadW(problem)};
+    }
+
+    CKArgs(const CKArgs&) = default;
+    CKArgs(CKArgs&&)      = default;
+    CKArgs& operator=(const CKArgs&) = default;
+
+    template <typename ConvPtr>
+    auto MakeArgPtr(const ConvPtr& conv_ptr,
+                    ConstData_t in_buf,
+                    ConstData_t w_buf,
+                    ConstData_t bias_buf,
+                    Data_t out_buf,
+                    float alpha,
+                    float beta,
+                    OutElementOp clampOp) const
+    {
+        (void)alpha;
+        (void)beta;
+
+        return conv_ptr->MakeArgumentPointer(
+            in_buf,
+            w_buf,
+            {bias_buf},
+            out_buf,
+            input_len,
+            in_strides,
+            weight_len,
+            wei_strides,
+            {output_len}, // hack CK's is applicable check. instead of bias_len we use output_len
+            {bias_strides},
+            output_len,
+            out_strides,
+            strides,
+            dilation,
+            lPadding,
+            rPadding,
+            in_element_op,
+            wei_element_op,
+            clampOp);
+    }
+
+    template <typename DevOpPtr>
+    auto MakeArgPtr(const DevOpPtr& op_ptr,
+                    const miopen::fusion::FusionInvokeParams& data_ctx) const
+    {
+        const auto& conv_param =
+            dynamic_cast<miopen::fusion::ConvolutionOpInvokeParam&>(*data_ctx.op_args.params[0]);
```

Comment:
In an event where we have a `dynamic_cast<ConvolutionOpInvokeParam&>` fail, it might not get to the assert part in a situations where ` params[0] ` is not a ConvolutionOpInvokeParam. Do you think using a pointer here instead, like `dynamic_cast<ConvolutionOpInvokeParam*>` along with the checks/exceptions, would be a better suitable alternative for this PR? 

This is can also be applicable to other parts in the code as well - where `dynamic_cast<>` is used.